.github/workflows/build.yml

```diff
@@ -29,24 +29,56 @@ jobs:
         with:
           username: ${{ secrets.DOCKERHUB_USERNAME }}
           password: ${{ secrets.DOCKERHUB_PASSWORD }}
+
       - name: Login to GitHub Container Registry
         uses: docker/login-action@v3
         with:
           registry: ghcr.io
           username: ${{ github.repository_owner }}
           password: ${{ secrets.GITHUB_TOKEN }}
 
-      - name: Build and push
+      - name: Get postfix of latest
+        id: postfix_of_latest
+        run: |
+          if [[ $GITHUB_REF_TYPE == tag || $GITHUB_REF_NAME == master ]]; then
+            echo "Postfix of latest: ''"
+            echo "postfix_of_latest=" >> $GITHUB_OUTPUT
+          else
+            echo "Postfix of latest: -$GITHUB_REF_NAME"
+            echo "postfix_of_latest=-$GITHUB_REF_NAME" >> $GITHUB_OUTPUT
+          fi
+
+      - name: Build and export
         uses: docker/build-push-action@v5
         with:
           tags: |
-            docker.io/fnndsc/chris_ui:latest
-            docker.io/fnndsc/chris_ui:${{ steps.info.outputs.version_string }}
-            ghcr.io/fnndsc/chris_ui:latest
-            ghcr.io/fnndsc/chris_ui:${{ steps.info.outputs.version_string }}
+            chris_ui:latest${{ steps.postfix_of_latest.outputs.postfix_of_latest}}
           context: .
           file: ./Dockerfile
           platforms: linux/amd64
+          push: false
+          cache-from: type=gha
+          cache-to: type=gha,mode=max
+          outputs: type=docker,dest=/tmp/chris_ui:latest${{ steps.postfix_of_latest.outputs.postfix_of_latest}}.tar
```

Comment:
I have no doubt that this works, but could it be simpler?

There is no need to write the image to a `.tar` then run `docker load ...` There are two approaches which I would prefer.

If we needed to build inside a container, then my preferred solution would be to run a local registry like this:

https://github.com/FNNDSC/microminc/blob/cf6f34685f0a69d933b3a60cb75c7f77b7a17117/.github/workflows/build.yml#L41-L83

However, since ChRIS_ui is a pretty simple React.JS application, I think it would be easier and faster to simply use [`actions/setup-node`](https://github.com/actions/setup-node), build the JavaScript application without using Docker, then build the container image using the `Dockerfile_cp` file you have below.

Reply:
(done)